staging/src/k8s.io/apiserver/pkg/server/config.go

```diff
@@ -223,10 +223,10 @@ type Config struct {
 	// If not specify any in flags, then genericapiserver will only enable defaultAPIResourceConfig.
 	MergedResourceConfig *serverstore.ResourceConfig
 
-	// lifecycleSignals provides access to the various signals
+	// LifecycleSignals provides access to the various signals
 	// that happen during lifecycle of the apiserver.
 	// it's intentionally marked private as it should never be overridden.
```

Comment:
see the comment 

> it's intentionally marked private as it should never be overridden.

I remember other PR trying to make it public and (I guess it was @sttts ) expressing concerns

Reply:
The alternative would be to add a method returning individual signals (for the sake of this PR we need the ShutdownInitiated one).

That would work for use, because we don't want to overwrite it. Would that be ok?

Reply:
maybe we can expose some kind of `StopNotify() <-chan struct{} {  }`

Reply:
OK - I like it. Let me try to redo this.